### ISSUE TITLE
Doxygen should have an option to collapse inline namespaces (Origin: bugzilla #759298)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#5914]
status NEW severity *enhancement* in component *general* for *---*
Reported in version *1.8.10* on platform *Other*
Assigned to: Dimitri van Heesch

On 2015-12-10 13:00:20 +0000, Jonathan Wakely wrote:

> namespace foo {
>   inline namespace _v1 {
>     /// Some type
>     struct Bar { };
>   }
> }
> 
> This will be documented as foo::_v1::Bar but one of the primary uses of inline namespaces is to make them invisible to users, so the public API should be documented as foo::Bar (without the inline namespace).
> 
> It would be good to get a STRIP_INLINE_NAMESPACES Doxyfile tag to control whether inline namespaces should be shown in the output. For more flexibility a \command would allow it to be specified per-namespace (sometimes an inline namespace is used for composition, rather than versioning, and then documenting it is useful).

**Comment by jwakely** [comment]
We use Doxygen for the GNU C++ standard library (libstdc++) and we use inline namespaces for ABI versioning, so for example we have `std::_V2::conditional_variable_any` but the `_V2` namespace is an implementation details so our public API docs should only refer to this as `std::condition_variable_any`.

We need a way for the `_V2` namespace to be entirely ignored by Doxygen, so it doesn't show up as a nested namespace in `std` and so its contents are documented as though they appeared directly in namespace `std`.

**Comment by mosra** [comment]
For your particular case, wouldn't specifying 

```ini
PREDEFINED = \
    _GLIBCXX_BEGIN_NAMESPACE_VERSION= \
    _GLIBCXX_END_NAMESPACE_VERSION=
```

in the Doxyfile solve this? (That's what I would do, inline namespace being a macro in the headers is very convenient for that.)

**Comment by jwakely** [comment]
No, because those macros are already empty. The `_V2` namespace is unrelated to those macros.

**Comment by mosra** [comment]
Ah, right, sorry. Mixed that up with libc++ where it *is* defined like that.
